### PR TITLE
feat: implementar módulo de Anamnese do paciente (#49)

### LINKED ISSUE #49: Implementar Anamnese do paciente
## Descrição do problema ou necessidade

O prontuário fisioterapêutico precisa registrar a anamnese do paciente, contendo informações clínicas, histórico de saúde, queixa principal, limitações, hábitos e objetivos iniciais. Essas informações são essenciais para orientar o plano terapêutico e a condução das sessões de Pilates.

## Objetivo da atividade

Implementar o módulo de Anamnese vinculado ao paciente, permitindo registrar, consultar e atualizar as informações iniciais coletadas no atendimento fisioterapêutico.

## Escopo da implementação

- Criar entidade/modelo de Anamnese relacionada ao Paciente.
- Criar DTOs de request e response.
- Criar repository, service e controller REST.
- Implementar endpoints para:
  - Criar anamnese para um paciente.
  - Buscar anamnese por paciente.
  - Buscar anamnese por ID.
  - Atualizar anamnese.
- Definir campos mínimos:
  - Queixa principal.
  - Histórico de doenças.
  - Histórico de cirurgias.
  - Histórico de lesões.
  - Medicamentos em uso.
  - Alergias.
  - Nível de atividade física.
  - Restrições médicas.
  - Objetivos do paciente.
  - Observações gerais.
- Validar existência do paciente antes de criar a anamnese.
- Definir regra para permitir uma ou mais anamneses por paciente. Sugestão inicial: uma anamnese principal atualizável por paciente.

## Critérios de aceite

- Deve ser possível criar uma anamnese vinculada a um paciente existente.
- Não deve ser possível criar anamnese para paciente inexistente.
- Deve ser possível consultar a anamnese de um paciente.
- Deve ser possível atualizar as informações da anamnese.
- A resposta da API deve trazer os dados principais da anamnese e a referência do paciente.
- O sistema deve validar campos obrigatórios definidos para o cadastro.
- O endpoint deve seguir o padrão REST do projeto.

## Sugestão de validação/testes

- Teste unitário para criação de anamnese com paciente válido.
- Teste para erro ao criar anamnese com paciente inexistente.
- Teste para atualização de anamnese.
- Teste para busca por paciente.
- Teste para validação dos campos obrigatórios.
- Teste de integração do controller, se aplicável.

## Observações técnicas

- Avaliar se a anamnese deve possuir histórico/versionamento no futuro.
- Evitar sobrescrever dados sem registrar data de atualização.
- Utilizar relacionamento adequado com Paciente.
- Considerar campos textuais flexíveis para permitir evolução do formulário sem grande impacto inicial.

## Labels sugeridas

- `enhancement`
- `backend`
- `anamnese`
- `prontuario`

### PR DESCRIPTION
**Pull request by FabioCarlesso**
## Resumo

Implementa o módulo completo de Anamnese vinculado ao Paciente, conforme especificado na issue #49. O módulo permite registrar, consultar e atualizar as informações clínicas iniciais coletadas no atendimento fisioterapêutico.

## Motivo da mudança

O prontuário fisioterapêutico precisava registrar a anamnese do paciente — queixa principal, histórico de saúde, medicamentos, limitações, hábitos e objetivos — para orientar o plano terapêutico e a condução das sessões de Pilates.

## O que foi implementado

- **Entidade `Anamnese`**: relacionamento `@OneToOne` com `Paciente`; constraint `UNIQUE` em `paciente_id` (uma anamnese principal por paciente)
- **DTOs**: `AnamneseRequestDTO`, `AnamneseUpdateDTO`, `AnamneseResponseDTO` (records com factory method `from`)
- **`AnamneseRepository`**: consulta por `paciente_id` e verificação de existência
- **`AnamneseService`**: criar, buscar por ID, buscar por paciente e atualizar (atualização parcial)
- **`AnamneseController`**: 4 endpoints REST em `/anamneses`
- **Migração Flyway `V14`**: tabela `anamneses` com FK para `pacientes` e índice em `paciente_id`

## Endpoints

| Método | Rota | Retorno |
|---|---|---|
| POST | `/anamneses` | 201 / 400 / 404 / 409 |
| GET | `/anamneses/{id}` | 200 / 404 |
| GET | `/anamneses/paciente/{pacienteId}` | 200 / 404 |
| PUT | `/anamneses/{id}` | 200 / 400 / 404 |

## Testes executados

- `AnamneseServiceTest` — 9 testes unitários (Mockito): criação com paciente válido, erro paciente inexistente, conflito anamnese duplicada, busca por ID, busca por paciente, atualização parcial e completa
- `AnamneseControllerTest` — 10 testes `@WebMvcTest` + MockMvc: validação de campos obrigatórios, respostas 201/400/404/409, atualização
- Suíte completa: **235 testes, 0 falhas**

## Arquivos principais alterados

- `src/main/java/.../entity/Anamnese.java`
- `src/main/java/.../dto/AnamneseRequestDTO.java`, `AnamneseUpdateDTO.java`, `AnamneseResponseDTO.java`
- `src/main/java/.../repository/AnamneseRepository.java`
- `src/main/java/.../service/AnamneseService.java`
- `src/main/java/.../controller/AnamneseController.java`
- `src/main/resources/db/migration/V14__create_anamneses_table.sql`
- `src/test/java/.../service/AnamneseServiceTest.java`
- `src/test/java/.../controller/AnamneseControllerTest.java`
- `docs/context.md`

## Referência

Closes #49

https://claude.ai/code/session_017TNdiEsqZPHfrp1o4fdfao

---
_Generated by [Claude Code](https://claude.ai/code/session_017TNdiEsqZPHfrp1o4fdfao)_

### CODE REVIEW

File: src/main/java/com/carlesso/pilatesapi/dto/AnamneseUpdateDTO.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.carlesso.pilatesapi.dto;
+
+public record AnamneseUpdateDTO(
+        String queixaPrincipal,
```
Comment (FabioCarlesso) [inline src/main/java/com/carlesso/pilatesapi/dto/AnamneseUpdateDTO.java:7]: - Problem: `queixaPrincipal` and `objetivos` are mandatory on create, but the update DTO allows sending empty/blank strings for these fields.
- Why it matters: the service applies any non-null value, so a request like `{ "queixaPrincipal": "" }` can leave the anamnese in an invalid business state even though the DB only enforces `NOT NULL`.
- Suggested improvement: add validation that allows `null` for partial update but rejects blank values when present, e.g. `@Size(min = 1)`/custom validator, or validate in the service with `isBlank()` before applying the update. Also add controller tests for blank `queixaPrincipal` and `objetivos`.
- Severity: Medium

File: src/main/java/com/carlesso/pilatesapi/service/AnamneseService.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    private final PacienteRepository pacienteRepository;
+
+    public AnamneseService(AnamneseRepository anamneseRepository, PacienteRepository pacienteRepository) {
+        this.anamneseRepository = anamneseRepository;
+        this.pacienteRepository = pacienteRepository;
+    }
+
+    @Transactional
+    public AnamneseResponseDTO criar(AnamneseRequestDTO dto) {
+        Paciente paciente = pacienteRepository.findById(dto.pacienteId())
```
Comment (FabioCarlesso) [inline src/main/java/com/carlesso/pilatesapi/service/AnamneseService.java]: - Problem: the creation uses `findById`, which allows creating an anamnese for an inactive patient.
- Why it matters: the project already treats inactive patients as soft-deleted in several business flows, and the documentation says aula queries filter `paciente.ativo = true`. Since anamnese contains clinical/sensitive data, allowing creation/lookup for inactive patients can expose or mutate records that should probably be out of the active workflow.
- Suggested improvement: confirm the expected rule. If anamneses should only apply to active patients, use a repository method such as `findByIdAndAtivoTrue`, and mirror that in `buscarPorId`/`buscarPorPaciente` using joins or repository queries that filter `paciente.ativo = true`. Add tests for inactive patient scenarios.
- Severity: Medium

File: src/main/java/com/carlesso/pilatesapi/service/AnamneseService.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                .orElseThrow(() -> new ResourceNotFoundException("Paciente não encontrado: " + dto.pacienteId()));
+
```
Comment (FabioCarlesso) [inline src/main/java/com/carlesso/pilatesapi/service/AnamneseService.java:33]: - Problem: the duplicate check with `existsByPacienteId` is not enough under concurrent requests.
- Why it matters: two requests for the same `pacienteId` can both pass this check before either insert commits. The database unique constraint will protect consistency, but the API may return a generic `DataIntegrityViolationException` response instead of the intended domain-specific conflict message.
- Suggested improvement: keep the DB unique constraint, but also catch/translate `DataIntegrityViolationException` around `save` into the same `ConflictException`, or rely on a centralized handler with a clear 409 message for this constraint. Add a test for duplicate constraint handling if possible.
- Severity: Medium

**Review by FabioCarlesso — COMMENTED** [review]
Review concluído.

Resumo por área:

1. Documentation: looks good. `docs/context.md` foi atualizado com estrutura, tabela `anamneses`, endpoints, regras de negócio e testes. Apenas recomendo alinhar a documentação caso seja adotada a regra de não permitir anamnese para paciente inativo.

2. Bugs and Edge Cases: encontrei alguns pontos de atenção: atualização permite strings em branco para campos obrigatórios; criação pode retornar erro genérico em cenário concorrente; e o service não diferencia paciente inativo de paciente válido.

3. Security: autenticação global parece cobrir as rotas de negócio conforme documentação do projeto. O principal cuidado é por se tratar de dados clínicos/sensíveis: evitar retorno indevido de anamnese de paciente inativo e avaliar autorização por perfil/escopo no futuro, se o sistema tiver múltiplos tipos de usuário.

4. Performance: fluxo simples e sem N+1 relevante. Há uma pequena oportunidade de reduzir uma consulta em `buscarPorPaciente`, mas não considero bloqueante.

5. Spring Boot Best Practices: estrutura controller/service/repository/DTO está consistente com o projeto. Sugiro melhorar validações do DTO de update e tratar exceção de constraint única de forma explícita.

6. Tests: bons testes de service e controller. Faltam casos para update com `blank`, criação concorrente/constraint única e comportamento com paciente inativo caso essa regra seja esperada.

7. Code Quality: implementação está legível, separada por camadas e consistente com a estrutura existente. Os comentários inline indicam os ajustes recomendados.

File: src/main/resources/db/migration/V14__create_anamneses_table.sql [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    medicamentos_uso      TEXT,
+    alergias              TEXT,
+    nivel_atividade_fisica VARCHAR(50),
+    restricoes_medicas    TEXT,
+    objetivos             TEXT            NOT NULL,
+    observacoes           TEXT,
+    data_criacao          TIMESTAMP       NOT NULL,
+    data_atualizacao      TIMESTAMP
+);
+
```
Comment (FabioCarlesso) [inline src/main/resources/db/migration/V14__create_anamneses_table.sql]: - Problem: `paciente_id` is already declared `UNIQUE`, and PostgreSQL creates an index for a unique constraint automatically. The explicit `idx_anamneses_paciente_id` duplicates that index.
- Why it matters: it adds unnecessary storage and extra maintenance cost on inserts/updates without improving lookups by `paciente_id`.
- Suggested improvement: remove the explicit index, or replace the inline `UNIQUE` with a named unique constraint/index if you want a predictable name.
- Severity: Low

**Comment by FabioCarlesso** [comment]
Ajustes da revisão aplicados no commit b43b1b7.

Resumo:
- Update de anamnese agora rejeita `queixaPrincipal` e `objetivos` em branco quando os campos são enviados, mantendo `null` permitido para atualização parcial.
- Criação, consulta e atualização de anamnese agora consideram apenas pacientes ativos.
- Violação concorrente da constraint única de `paciente_id` é traduzida para `ConflictException`/409.
- Índice explícito duplicado em `anamneses(paciente_id)` foi removido da migration, mantendo o índice automático da constraint `UNIQUE`.
- Documentação atualizada com as regras de paciente ativo e índice automático.

Validação executada:
- `mvn test -Dtest=AnamneseServiceTest,AnamneseControllerTest` — 31 testes, 0 falhas
- `mvn test` — 243 testes, 0 falhas
